src/Plugin/CheckImports.hs

```diff
@@ -91,10 +101,16 @@ typecheckPlugin _ modsummary tc = do
       moduleImportMap =
         foldl' (\(!m) (modu, name) -> M.insertWith S.union modu (S.singleton name) m) M.empty $
           concatMap mkModuleNameMap usedGREs
-  for_ (M.toList moduleImportMap) $ \(modu, names) -> liftIO $ do
-    putStrLn "---------"
-    printPpr dflags modu
-    let imported = fmap (formatName dflags) $ S.toList names
-    putStrLn $ formatImportedNames imported
+
+  let rendered =
+        unlines $
+          flip concatMap (M.toList moduleImportMap) $ \(modu, names) ->
```

Comment:
`flip concatMap` is also `>>=` - may be cleaner to write as `do` notation, too

Reply:
done, thanks!